[CI] data-streams/80_resolve_index_data_streams failure

Test fails, because ilm history index and alias exist and the test doesn't take this into account.

**Build scan**: https://gradle-enterprise.elastic.co/s/g3357d3mch5kk/tests/:x-pack:plugin:data-streams:qa:rest:integTest/org.elasticsearch.xpack.datastreams.DataStreamsRestIT/test%20%7Bp0=data-streams%2F80_resolve_index_data_streams%2FResolve%20index%20with%20hidden%20and%20closed%20indices%7D#1

**Repro line**: ./gradlew ':x-pack:plugin:data-streams:qa:rest:integTest' --tests "org.elasticsearch.xpack.datastreams.DataStreamsRestIT.test {p0=data-streams/80_resolve_index_data_streams/Resolve index with hidden and closed indices}" -Dtests.seed=C8055B33FA25C7BC -Dtests.security.manager=true -Dtests.locale=de-CH -Dtests.timezone=Pacific/Chatham -Druntime.java=8

**Reproduces locally?**: No

**Applicable branches**: master, 7.x, 7.9

**Failure history**: https://gradle-enterprise.elastic.co/scans/tests?search.buildToolTypes=gradle&search.buildToolTypes=maven&search.relativeStartTime=P7D&search.timeZoneId=Europe/Amsterdam&tests.container=org.elasticsearch.xpack.datastreams.DataStreamsRestIT&tests.sortField=FAILED&tests.test=test%20%7Bp0%3Ddata-streams/80_resolve_index_data_streams/Resolve%20index%20with%20hidden%20and%20closed%20indices%7D&tests.unstableOnly=true

**Failure excerpt**:

```
Caused by: java.lang.AssertionError: indices.3.name didn't match expected value: |  
  | indices.3.name: expected String [test_index1] but was String [ilm-history-2-000001]
```

```
[2020-08-19T02:26:25,159][INFO ][o.e.x.d.DataStreamsRestIT] [test] Stash dump on test failure [{ |  
-- | --
  | "stash" : { |  
  | "body" : { |  
  | "indices" : [ |  
  | { |  
  | "name" : ".ds-simple-data-stream1-000001", |  
  | "attributes" : [ |  
  | "hidden", |  
  | "open" |  
  | ], |  
  | "data_stream" : "simple-data-stream1" |  
  | }, |  
  | { |  
  | "name" : ".ds-simple-data-stream2-000001", |  
  | "attributes" : [ |  
  | "hidden", |  
  | "open" |  
  | ], |  
  | "data_stream" : "simple-data-stream2" |  
  | }, |  
  | { |  
  | "name" : ".ds-simple-data-stream2-000002", |  
  | "attributes" : [ |  
  | "hidden", |  
  | "open" |  
  | ], |  
  | "data_stream" : "simple-data-stream2" |  
  | }, |  
  | { |  
  | "name" : "ilm-history-2-000001", |  
  | "aliases" : [ |  
  | "ilm-history-2" |  
  | ], |  
  | "attributes" : [ |  
  | "hidden", |  
  | "open" |  
  | ] |  
  | }, |  
  | { |  
  | "name" : "test_index1", |  
  | "aliases" : [ |  
  | "test_alias", |  
  | "test_blias", |  
  | "test_clias" |  
  | ], |  
  | "attributes" : [ |  
  | "open" |  
  | ] |  
  | }, |  
  | { |  
  | "name" : "test_index2", |  
  | "aliases" : [ |  
  | "test_alias" |  
  | ], |  
  | "attributes" : [ |  
  | "closed" |  
  | ] |  
  | }, |  
  | { |  
  | "name" : "test_index3", |  
  | "aliases" : [ |  
  | "test_blias" |  
  | ], |  
  | "attributes" : [ |  
  | "open" |  
  | ] |  
  | } |  
  | ], |  
  | "aliases" : [ |  
  | { |  
  | "name" : "ilm-history-2", |  
  | "indices" : [ |  
  | "ilm-history-2-000001" |  
  | ] |  
  | }, |  
  | { |  
  | "name" : "test_alias", |  
  | "indices" : [ |  
  | "test_index1", |  
  | "test_index2" |  
  | ] |  
  | }, |  
  | { |  
  | "name" : "test_blias", |  
  | "indices" : [ |  
  | "test_index1", |  
  | "test_index3" |  
  | ] |  
  | }, |  
  | { |  
  | "name" : "test_clias", |  
  | "indices" : [ |  
  | "test_index1" |  
  | ] |  
  | } |  
  | ], |  
  | "data_streams" : [ |  
  | { |  
  | "name" : "simple-data-stream1", |  
  | "backing_indices" : [ |  
  | ".ds-simple-data-stream1-000001" |  
  | ], |  
  | "timestamp_field" : "@timestamp" |  
  | }, |  
  | { |  
  | "name" : "simple-data-stream2", |  
  | "backing_indices" : [ |  
  | ".ds-simple-data-stream2-000001", |  
  | ".ds-simple-data-stream2-000002" |  
  | ], |  
  | "timestamp_field" : "@timestamp" |  
  | } |  
  | ] |  
  | } |  
  | } |  
  | }]
```

--- a/x-pack/plugin/data-streams/qa/rest/build.gradle
+++ b/x-pack/plugin/data-streams/qa/rest/build.gradle
@@ -17,4 +17,6 @@ dependencies {
 testClusters.integTest {
   testDistribution = 'DEFAULT'
   setting 'xpack.license.self_generated.type', 'basic'
+  // disable ILM history, since it disturbs tests using _all
+  setting 'indices.lifecycle.history_index_enabled', 'false'
 }

